### ISSUE TITLE
A question about this library

### ISSUE THREAD

**Issue by visortelle** [hiroki0525/idle-task#153]
Hi 👋

**TLDR:** Does it run in inactive tab?

---

I'm looking for a solution to the following problem:

Before closing a tab, I need to send an API request from the browser to server to the make a graceful teardown of resources allocated on the server side for the specific browser tab. \
I used `beforeunload` event for that, but this event doesn't emit when the user closes the tab when they are on another tab.
The solution that comes to my mind is to implement some kind of periodic heartbeat from browser to server.

My first thought was to implement it using Web Workers running in the background. After a bit more thinking on this option, it looks like it won't solve my problem.

I wonder, can this library help me to solve my task?

I'm not familiar with the `requestIdleCallback` browser API.

**Comment by hiroki0525** [comment]
Hi 👋

The answer is probably yes, but there might be some limitations.

Browsers like Chrome may take actions such as limiting the execution of JavaScript, including `requestIdleCallback`, on inactive tabs to reduce memory and CPU usage.

However, based on the experiments I conducted, it seemed that `requestIdleCallback` was still being executed on inactive tabs.

You may be able to verify that `requestIdleCallback` is being executed by running the following code, switching to another tab, and then returning to the tab where the code was executed.

```js
let count = 0;

function test() {
  requestIdleCallback(() => {
    count++;
    console.log(count);
    test();
  });
}

test();
```

**Comment by visortelle** [comment]
@hiroki0525 Thank you for the answer!

I replaced count with `Date.now()` and ran it in a background tab for half an hour.
Unfortunately, it doesn't work as want 🤣

It seems I initially misunderstood what `requestIdleCallback` does and it isn't intended to solve my problem.

**Comment by dbalatero** [comment]
@visortelle @hiroki0525 If you pass `timeout` to `requestIdleCallback`, that apparently helps the browser know that all work is required – otherwise it's at the browser's discretion to defer the entire idle queue until the user comes back to that tab.

See this article for a breakdown: https://eric-schaefer.com/til/2023/03/11/the-dark-side-of-requestidlecallback/
